pkg/operator/tektonaddons.go

```diff
@@ -143,3 +145,37 @@ func verifyNoTektonAddonCR(clients *clients.Clients) error {
 	}
 	return nil
 }
+
+// VerifyVersionedTasks checks if the required tasks are available with the expected version
+func VerifyVersionedTasks(taskList string, requiredTasks []string) {
+	getRequiredVersion := os.Getenv("OSP_VERSION")
+	// Get the arch of the cluster as kn and Kn-apply task are not available on arm64 cluster
+	if config.Flags.ClusterArch == "arm64" {
+		updatedTasks := []string{}
+		for _, task := range requiredTasks {
+			if task != "kn" && task != "kn-apply" {
+				updatedTasks = append(updatedTasks, task)
+			}
+		}
+		requiredTasks = updatedTasks
+	}
```

Comment:
Wouldn't it be easier if `requiredTasks` contained all but `kn` and `kn-apply` and you would add them inside `if not arm64`?

Reply:
Done